src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.SnapPoints.cs

```diff
@@ -385,32 +385,22 @@ private void AdjustOffsetsForSnapPoints(ref double? horizontalOffset, ref double
 						// it doesn't matter so do it for consistency.
 						snapPoint *= zoomFactor;
 
-						var alignedSnapPoint = alignment switch
+						// Adjust snap point to relative(near/center/far) viewport offset, and then
+						// clamp it within valid scroll range.
+						var adjustedOffset = alignment switch
 						{
 							SnapPointsAlignment.Near => snapPoint,
 							SnapPointsAlignment.Center => (float)(snapPoint - viewportDimension / 2),
 							SnapPointsAlignment.Far => (float)(snapPoint - viewportDimension),
 							_ => throw new IndexOutOfRangeException("alignment")
 						};
+						var clampedOffset = (float)Math.Max(0, Math.Min(adjustedOffset, ScrollableWidth));
 
-						var distanceFromFarEdge = (float)(extentDimension - viewportDimension - alignedSnapPoint);
-
-						// With certain scale factors and resolutions the snap points sometimes get pushed out of bounds
-						// by rounding errors, in these cases we need to intervene and return them to the boundary.
-						if (distanceFromFarEdge < 0 &&
-						    distanceFromFarEdge >= (-1.0 * (ScrollViewerSnapPointLocationTolerance * Math.Max(1.0f, zoomFactor))))
-						{
-							alignedSnapPoint = (float)(extentDimension - viewportDimension);
-							distanceFromFarEdge = 0;
-						}
-						if (alignedSnapPoint >= 0 && distanceFromFarEdge >= 0)
+						if (staticZoomFactor == 1.0f)
 						{
-							if (staticZoomFactor == 1.0f)
-							{
-								alignedSnapPoint /= zoomFactor;
-							}
-							result.Add(alignedSnapPoint);
```

Comment:
This code is shared, won't this have an impact on Skia ? @dr1rrb

Reply:
I'm also not seeing it anywhere else. Is it not used at all ?

Reply:
I would agree. Also, snap points are not a feature specific to `ListView`, is it going to work?

Reply:
Indeed, this change here affects any-plat... It is call when the ScrollViewer is scrolling (manually, programmatically or from animation?):
```
CopyMotionSnapPoints // we are here
    GetScrollSnapPoints
        AdjustOffsetWithMandatorySnapPoints
            AdjustOffsetsForSnapPoints
                ChangeViewCore
                DelayedMoveToSnapPoint
```
---
There are two part to this code:
a. first if-block: clamping the result `alignedSnapPoint` past the edge against the edge (with compensation for scaling+rounding error)
b. 2nd if-block: select result that is within valid bounds

I chomped off the first block to replaced by `Math.Min(0, Math.Max(..., ScrollableWidth))` (ScrollableWidth is extentDimension - viewportDimension; In retrospect, float-float is a lot better than accessing a DP).
I also changed the 2nd block, because it was rejecting the snap-points for the 1st and the last element in the ScrollViewer, instead it will now accepted any clamped value as result.
I think these changes make sense.

And, here are the before and after comparison on a specific scenario: // generated with: http://share.linqpad.net/pr48jr.linq
// extend=1080, view-port=411, item-size=360
![image](https://user-images.githubusercontent.com/2359550/203397650-e186fddd-f89b-4a30-816a-2d7c8cbbbab8.png)
![image](https://user-images.githubusercontent.com/2359550/203397981-0dceac8f-a3c1-4d3d-b3c2-6b8b6fb39fed.png)